In [42]:
import pandas as pd
import numpy as np
import seaborn as sns

In [36]:
df=pd.read_csv('heights.csv')
df

gender     height
0       Male  73.847017
1       Male  68.781904
2       Male  74.110105
3       Male  71.730978
4       Male  69.881796
...      ...        ...
9995  Female  66.172652
9996  Female  67.067155
9997  Female  63.867992
9998  Female  69.034243
9999  Female  61.944246

[10000 rows x 2 columns]

In [51]:
Z_Score=(df['height'] - df['height'].mean())/df['height'].std()
Z_Score

0       1.943964
1       0.627505
2       2.012343
3       1.393991
4       0.913375
          ...   
9995   -0.050658
9996    0.181830
9997   -0.649655
9998    0.693090
9999   -1.149651
Name: height, Length: 10000, dtype: float64

Z Score adalah seberapa jauh standard deviasi dari mean dataset

In [52]:
df['Z-Score']=Z_Score
df

gender     height   Z-Score
0       Male  73.847017  1.943964
1       Male  68.781904  0.627505
2       Male  74.110105  2.012343
3       Male  71.730978  1.393991
4       Male  69.881796  0.913375
...      ...        ...       ...
9995  Female  66.172652 -0.050658
9996  Female  67.067155  0.181830
9997  Female  63.867992 -0.649655
9998  Female  69.034243  0.693090
9999  Female  61.944246 -1.149651

[10000 rows x 3 columns]

Ini berarti pada nilai pertama,yaitu Z-Score dengan 1.94 berarti std nya sejauh 1.94 dari mean dataset

In [55]:
mean_per_height=df['height'].mean()
mean_per_height

66.367559754866

In [56]:
std_per_height=df['height'].std()
std_per_height

3.847528120795573

In [61]:
std_dataset=(df['height'][0] - mean_per_height)/std_per_height
std_dataset

1.9439642883201165

In [63]:
df[df['Z-Score']>3]

gender     height   Z-Score
994    Male  78.095867  3.048271
1317   Male  78.462053  3.143445
2014   Male  78.998742  3.282934
3285   Male  78.528210  3.160640
3757   Male  78.621374  3.184854

In [64]:
df[df['Z-Score']<-3]

gender     height   Z-Score
6624  Female  54.616858 -3.054091
9285  Female  54.263133 -3.146027